Dev mode crashes with FileNotFoundException on fresh workspace (build-metrics.json)

## Problem

When running `bazel run helloworld_dev` on a fresh workspace, Quarkus dev mode crashes immediately:

```
java.lang.RuntimeException: java.io.FileNotFoundException: 
  .../examples/helloworld_3_33/target/build-metrics.json (No such file or directory)
    at io.quarkus.runner.bootstrap.AugmentActionImpl.runAugment(AugmentActionImpl.java:429)
    ...
Caused by: java.io.FileNotFoundException: .../target/build-metrics.json (No such file or directory)
    at io.quarkus.builder.BuildMetrics.dumpTo(BuildMetrics.java:194)
    at io.quarkus.deployment.QuarkusAugmentor.run(QuarkusAugmentor.java:192)
```

## Root cause

`DevModeLauncher.buildDevModeContext()` sets `targetDir` to `projectRoot.resolve("target")`. In a Maven project this directory always exists, but in a Bazel workspace it doesn't — there's no `target/` directory by default.

Quarkus's `QuarkusAugmentor` writes `build-metrics.json` into `targetDir`, and `WorkspaceProcessor` derives the project root from `targetDir.getParent()` for the Dev UI Workspace tab. So `targetDir` must:
1. Be a child of the workspace root (so `.getParent()` gives the correct project root)
2. Actually exist on disk (so `build-metrics.json` can be written)

## Fix

In `DevModeLauncher.launch()`, call `Files.createDirectories()` on the resolved `targetDir` before Quarkus starts. This ensures the directory exists while preserving the correct parent relationship for the Workspace tab.

## Repro

```bash
cd examples/helloworld_3_33
bazel run helloworld_dev
```

--- a/quarkifier/src/main/java/com/clementguillot/quarkifier/dev/DevModeLauncher.java
+++ b/quarkifier/src/main/java/com/clementguillot/quarkifier/dev/DevModeLauncher.java
@@ -57,6 +57,10 @@ public final class DevModeLauncher {
     try {
       DevModeContext context = buildDevModeContext(config);
 
+      // Ensure the target directory exists — Quarkus writes build-metrics.json there.
+      // In a Bazel workspace this directory doesn't exist by default (unlike Maven's target/).
+      Files.createDirectories(Path.of(context.getApplicationRoot().getTargetDir()));
+
       // 1. Serialize the ApplicationModel to a temp file.
       // Uses AppModelSerializerImpl which is version-specific:
       // - 3.27: Java Object Serialization (BootstrapUtils)
@@ -253,6 +257,11 @@ public final class DevModeLauncher {
     // Key: classesPath points to mutable directory when available, otherwise the jar
     Path classesPath = config.classesDir() != null ? config.classesDir() : appJar;
 
+    // targetDir must be a child of projectRoot so that WorkspaceProcessor (which does
+    // targetDir.getParent() to find the project root) shows the correct source tree.
+    // We create the directory in launch() since Bazel workspaces don't have a target/ dir.
+    Path targetDir = projectRoot.resolve("target");
+
     var moduleInfo =
         new DevModeContext.ModuleInfo.Builder()
             .setArtifactKey(ArtifactKey.ga(coords.groupId(), coords.artifactId()))
@@ -265,7 +274,7 @@ public final class DevModeLauncher {
                 config.resources().isEmpty()
                     ? null
                     : config.resources().get(0).toAbsolutePath().toString())
-            .setTargetDir(projectRoot.resolve("target").toAbsolutePath().toString())
+            .setTargetDir(targetDir.toAbsolutePath().toString())
             .build();
 
     context.setApplicationRoot(moduleInfo);

